ChatApplication/Services/MessageService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using ChatApplication.Data;
+using ChatApplication.Models;
+using ChatApplication.Services.Interfaces;
+using ChatApplication.ViewModels;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace ChatApplication.Services
+{
+    public class MessageService : IMessageService
+    {
+        private readonly ICacheExtensionsService _cacheExtensionsService;
+        private readonly ApplicationDbContext _context;
+        private readonly UserManager<AppUser> _userManager;
+
+        public MessageService(ApplicationDbContext context, ICacheExtensionsService cacheExtensionsService, UserManager<AppUser> userManager)
+        {
+            _context = context;
+            _cacheExtensionsService = cacheExtensionsService;
+            _userManager = userManager;
+        }
+
+        public async Task<IEnumerable<RedisCacheDataModel>> GetCacheData()
+        {
+            var messages = await _cacheExtensionsService.GetCacheValueAsync();
+            var messageList = messages.ToList();
+
+            return messageList;
+        }
+
+        public async Task AddDbData(ClaimsPrincipal user, string text)
+        {
+            var sender = await _userManager.GetUserAsync(user);
```

Comment:
Why did you fetch user info from identity in the service layer, in case you have already "ClaimsPrincipal user". If I am not mistaken you have read about that already.